libbeat/processors/add_cloud_metadata/add_cloud_metadata.go

```diff
@@ -358,8 +363,14 @@ func (p *addCloudMetadata) Run(event *beat.Event) (*beat.Event, error) {
 		return event, nil
 	}
 
-	// This overwrites the meta.cloud if it exists. But the cloud key should be
-	// reserved for this processor so this should happen.
+	// If cloud key exists in event already and overwrite flag is set to false, this processor will not overwrite the
+	// cloud fields. For example aws module writes cloud.instance.* to events already, with overwrite=false,
+	// add_cloud_metadata should not overwrite these fields with new values.
+	cloudValue, _ := event.GetValue("cloud")
```

Comment:
You should surround the entire `GetValue` block in an `if !overwrite` check to avoid doing the work of fetching the `cloud` value when it’s just going to write over it.